ci/github-actions/fuzz/action.yml

```diff
@@ -63,12 +67,23 @@ runs:
         echo "ARCHIVE_PATH=go-ci-fuzz_${ARCHIVE_OS}_${ARCHIVE_ARCH}.${ARCHIVE_EXT}" >> "$GITHUB_OUTPUT"
         echo "EXE_NAME=go-ci-fuzz${EXE_EXT}" >> "$GITHUB_OUTPUT"
     - id: fetch-asset
-      name: Get Release Asset
-      uses: dsaltares/fetch-gh-release-asset@3942ce82f1192754cd487a86f03eef6eeb89b5da # 1.1.0
-      with:
-        repo: 'form3tech-oss/go-ci-fuzz'
-        version: ${{ inputs.version == 'latest' && 'latest' || format('tags/v{0}', inputs.version) }}
-        file: ${{ steps.archive-info.outputs.ARCHIVE_PATH }}
+      name: Download Go Ci Fuzz release
+      shell: bash
+      run: |
+        set -euo pipefail
+        FILE_NAME=${{ steps.archive-info.outputs.ARCHIVE_PATH }}
+        REL_PATH=${{ inputs.version == 'latest' && 'latest' || format('tags/v{0}', inputs.version) }}
+        RELEASE=$(curl -H "Accept: application/vnd.github+json" "https://api.github.com/repos/form3tech-oss/go-ci-fuzz/releases/$REL_PATH")
+        DOWNLOAD_PATH=$(<<<$RELEASE jq -r --arg file_name "$FILE_NAME" '.assets[] | select (.name == $file_name) | .browser_download_url')
+        curl -L -o "$FILE_NAME" "$DOWNLOAD_PATH"
+        
+        if [ ! -z "${{ inputs.checksum }}" ]; then
+          echo "${{ inputs.checksum }} $FILE_NAME" | sha256sum -c
+          if [ $? != 0 ]; then
```

Comment:
Is it worth cleaning up the downloaded artefact in case the checksum didn't match? :thinking: 
Just to avoid it being accidentally used further down the line.

Reply:
I checked [the logs](https://github.com/form3tech-oss/go-ci-fuzz/actions/runs/8421948912/job/23060063802) and since we have `-e -o pipefail` this `if` will never be reached. It's kind of a defensive approach.

We could 
- leave it as is
- unset `-e -o pipefail` opts and let the if handle the failure and remove the binary.
- remove the `if` (not deleting the binary)

What do you think?

Reply:
It's probably not a big deal, as only the last step has `if: 'failure()'` so nothing else would run anyway.
I'd say I have slight preference towards the 2nd option, but happy to leave as is.